infrastructure/builder/src/generate/generators/k8s.js

```diff
@@ -0,0 +1,354 @@
+      return {
+        [`ingresses-${name}`]: await renderTemplates(name, vars, procs, templates),
+        [`procslist-${name}`]: procs,
+      };
+    },
+  });
+});
+
+// Now add ui/references separately
+const extras = {
+  ui: {
+    vars: [],
+    procs: {
+      web: {
+        type: 'web',
+        readinessPath: '/',
+        paths: [
+          '/*',
+        ],
+      },
+    },
+  },
+  references: {
+    vars: [],
+    procs: {
+      web: {
+        type: 'web',
+        readinessPath: '/references/',
+        paths: [
+          '/references',
+          '/references/*',
+          '/schemas',
+          '/schemas/*',
+        ],
+      },
+    },
+  },
+};
+Object.entries(extras).forEach(([name, {procs, vars}]) => {
+  exports.tasks.push({
+    title: `Generate helm templates for ${name}`,
+    requires: ['k8s-templates'],
+    provides: [`ingresses-${name}`],
+    run: async (requirements, utils) => {
+      const templates = requirements['k8s-templates'];
+      return {
+        [`ingresses-${name}`]: await renderTemplates(name, vars.map(v => v.var), procs, templates),
+      };
+    },
+  });
+});
+
+exports.tasks.push({
+  title: `Generate ingress`,
+  requires: ['k8s-templates', 'ingresses-ui', 'ingresses-references', ...SERVICES.map(name => `ingresses-${name}`)],
+  provides: [],
+  run: async (requirements, utils) => {
+    const ingresses = [];
+    for (const [name, req] of Object.entries(requirements)) {
+      if (name.startsWith('ingresses-')) {
+        for (const ingress of req) {
+          for (const path of ingress.paths) {
+            ingresses.push({
+              path,
+              projectName: ingress.projectName,
+            });
+          }
+        }
+      }
+    }
+    const templates = requirements['k8s-templates'];
+    const rendered = jsone(templates['ingress'], {ingresses});
+    await writeRepoYAML(path.join(TMPL_DIR, 'ingress.yaml'), rendered);
+  },
+});
+
+exports.tasks.push({
+  title: `Generate values.yaml and values.schema.yaml`,
+  requires: [...SERVICES.map(name => `configs-${name}`), ...SERVICES.map(name => `procslist-${name}`)],
+  provides: [],
+  run: async (requirements, utils) => {
+    const schema = {
+      '$schema': 'http://json-schema.org/draft-06/schema#',
+      type: 'object',
+      title: 'Taskcluster Configuration Values',
+      properties: {
+        rootUrl: {
+          type: 'string',
+          format: 'uri',
+          description: 'The url pointing to your cluster ingress.',
+        },
+        dockerImage: {
+          type: 'string',
+          description: 'The docker image containing taskcluster.',
+        },
+        ingressStaticIpName: {
+          type: 'string',
+          description: 'A google static ip object that the ingress can use to maintain the same ip address.',
+        },
+        ingressCertName: {
+          type: 'string',
+          description: 'A google certificate name that the ingress can use to set up tls.',
+        },
```

Comment:
Hm, more GCP-specific stuff..

Reply:
If it's better for you, you could switch back to nginx ingress here and we could add in the gcp ingress on the cloudops side.

We're eventually going to switch away from straight gclb=>taskcluster to gclb=>iprepd-nginx=>taskcluster, which wouldn't make sense to bother supporting here since no one else will want to use iprepd-nginx,

Reply:
I think in this PR we'll only support GKE and then we can come by in a later patch and make this generally useful.